workspaces/spectacle/src/index.ts

```diff
@@ -110,6 +114,19 @@ export interface IOpticContext {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+export type AsyncStatus<T> =
```

Comment:
do you know a better pattern for this?

Reply:
So this pattern allows us to check `!loading` and `!error` - > which typescript will assert and know that data is _non nullable_ `T` - this is a similar pattern that react-use uses https://github.com/streamich/react-use/blob/master/src/useAsyncFn.ts#L5-L25 which I think provides us better type safety.

Reply:
I think I need to modify this definition (to be more similar to react-use since my definition here doesn't include error types)

Reply:
cool. yeah i also notice we've duplicated it a bit in a few files